### DISCUSSION TITLE
FindAllMarkers()

### DISCUSSION THREAD

**Discussion by rythi4** [satijalab/seurat#5335] [Q&A]
I'm running FindAllMarkers() on a published dataset, where I downloaded the metadata and a normalized count matrix and made a seuratobject. 
I'm trying to use FindAllMarkers() to find positive markers, but I don't understand the results I get:

![image](https://user-images.githubusercontent.com/92912238/143574227-15aa3f89-ee94-4f97-b56f-7704b9b5ef24.png)

I don't understand how pct.1 can be lower than pct.2 and still be a positive marker, with a positive log2fold change.

I ran:
Markers <- FindAllMarkers(seurat_object, min.pct = 0.25, only.pos = TRUE)

**Comment by samuel-marsh** [comment]
Hi,

I believe `only.pos` refers to the log fold change values and not the percent expressed values.  So it appears to be functioning as expected.  You can always filter the returned dataframe after running `FindAllMarkers` for anything you want, including whether to only include samples that have higher pct.1.

Best,
Sam

**Reply by rythi4** [reply]
As I understand it, avg_log2FC is a comparison between the expression in the two groups. And therefore positive values indicate that there is more expression in group 1, and negative values indicate that there is more expression in group 2. Therefore, my initial thought was that when avg_log2FC is positive, then pct.1 > pct.2. 
But could the avg_log2FC be positive when pct.1 < pct.2, because the few cells in group 1 have very high expression of the gene in question?

**Reply by crazyxiaoj** [reply]
I also encountered a similar problem. I looked at the violin plot and was surprised to find that `pct.2` has a larger "belly" compared to `pct.1`, and it seems that `pct.2` has higher expression than `pct.1`. But then I looked at the dot plot and found that the dot plot showed a trend of higher expression in `pct.2`, while the trend in the violin plot was wrong. So you can try using a dot plot for confirmation?
